student_work/evan_connect_four.rb

```diff
@@ -0,0 +1,225 @@
+        #If it is not: Ask user to give new input
+        #Else: Enter move into array
+  end
+end
+
+# This is put in the same file below the other functions
+def game_over?(board)
+  check_for_win(board) || all_spots_taken?(board)
+end
+
+def check_for_win(board)
+  # Check rows for win
+  row_win(board) ||
+  # Check columns for win
+  col_win(board) ||
+  # check right slant diagonals for win
+  diag_right_win(board) ||
+  # check left slant diagonals for win
+  diag_left_win(board)
+end
+
+def row_win(board)
+  win = false
+  row = -1
+  while !win && row > -7
+    col = 0
+    while !win && col < 3
+      win = ((board[row][col] != nil) && (board[row][col] == board[row][col + 1]) &&
+          (board[row][col] == board[row][col + 2]) &&
+          (board[row][col] == board[row][col + 3]))
+      if !win
+        col += 1
+      end
+    end
+    row -= 1
+  end
+  return win
+end
+
+def col_win(board)
+  win = false
+  col = -1
+  while !win && col > -7
+    row = 0
+    while !win && row < 3
+      win = ((board[row][col] != nil) && (board[row][col] == board[row + 1][col]) &&
+          (board[row][col] == board[row + 2][col]) &&
+          (board[row][col] == board[row + 3][col]))
+      if !win
+        row += 1
+      end
+    end
+    col -= 1
+  end
+  return win
+end
+
+def diag_right_win(board)
+  win = false
+  row = 3
+  while !win && row < 6
+    col = 0
+    while !win && col < 3
+      win = ((board[row][col] != nil)&&(board[row][col] == board[row-1][col+1]) && (board[row][col] == board[row - 2][col + 2]) &&
+          (board[row][col] == board[row - 3][col + 3]))
+      if !win
+        col += 1
+      end
+    end
+    row += 1
+  end
+  return win
+end
+
+def diag_left_win(board)
+  win = false
+  row = 3
+  while !win && row < 6
+    col = -1
+    while !win && col > -4
+      win = ((board[row][col] != nil)&&(board[row][col] == board[row-1][col-1]) && (board[row][col] == board[row - 2][col - 2]) &&
+          (board[row][col] == board[row - 3][col - 3]))
+      if !win
+        col -= 1
+      end
+    end
+    row += 1
+  end
+  return win
+end
+
+def all_spots_taken?(board)
+  # Check that all spots have been taken
+  !board[0].include?(nil) &&
+  !board[1].include?(nil) &&
+  !board[2].include?(nil) &&
+  !board[3].include?(nil) &&
+  !board[4].include?(nil) &&
+  !board[4].include?(nil)
+end
+
+def space_available?(board, space)
+  playable = false
+  count = -1
+  while !playable && count > -7
+    if(board[count][space] == nil)
+      playable = true
+    else
+      count -= 1
+    end
+  end
+  return playable
+end
+
+def switch_player(player)
+  if player == "X"
+    player = "O"
+  else
+    player = "X"
+  end
+  return player
+end
+
+def get_value(value)
+  if value == nil
+    value = "   "
+  else
+    value = " " + value + " "
+  end
+  return value
+end
+
+# Structure of function
+def print_game_board(board)
+  # print first row of spaces
+  iter = 0
+  puts("-" * 25)
+  while iter != board.count
```

Comment:
this can be dangerous.  If you don't correctly increment iter it can create an infinite loop.  Better practice is to use `<`